platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_sfp.py

```diff
@@ -321,8 +333,20 @@ def set_lpmode(self, lpmode):
             except IOError as e:
                 status = False
         else:
-            # Use common SfpOptoeBase implementation for set_lpmode
-            status = super().set_lpmode(lpmode)
+            xcvr_id = self._xcvr_api_factory._get_id()
+            if xcvr_id is not None:
+                if xcvr_id == 0x18 or xcvr_id == 0x19 or xcvr_id == 0x1e:
+                    # QSFP-DD or OSFP
+                    # Use common SfpOptoeBase implementation for set_lpmode
+                    status = super().set_lpmode(lpmode)
+                elif xcvr_id == 0x11 or xcvr_id == 0x0d or xcvr_id == 0x0c:
+                    # QSFP28, QSFP+, QSFP
+                    val = self.read_eeprom(QSFP_PWR_CTRL_ADDR, 1)
+                    if val is not None:
+                        val = ord(val) & 0xf0
+                        val |= 0x3 if lpmode else 0xd
+                        buf = bytearray([val])
+                        status = self.write_eeprom(QSFP_PWR_CTRL_ADDR, 1, buf)
```

Comment:
use [set_power_override](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/sff8436.py#L253) for QSFP28/QSFP+. For eg see [here](https://github.com/sonic-net/sonic-buildimage/blob/master/device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py#L372)